### ISSUE TITLE
Fatal error: Uncaught SellingPartnerApi\ApiException: [400] { "errors": [ { "code": "InvalidInput", "message": "Application does not have access to one or more requested data elements: [buyerInfo]", "details": "" } ] } in E:\xampp\htdoc2\selling\lib\Api\TokensApi.php:150 Stack trace: #0 E:\xampp\htdoc2\selling\lib\Api\TokensApi.php(125)

### ISSUE THREAD

**Issue by WebDevSand** [amzn/selling-partner-api-docs#1488]
I think there is issue in adding x-amz-access-token to sp api header.

**Comment by mgli0129** [comment]
Even I refreshed the refresh-token, I still couldn't access the PII.

**Comment by martr84** [comment]
We had tried to get this working we went through the process of clicking edit app and ticking the DirectToConsumer Role after being approved for the role and then regenerating the refresh token. however that DID NOT work.

What DID work is creating a brand new app registration with the same role arn and ticking all the required roles as part of the initial setup. This gave us new lwa credentials, new app ids and new refresh tokens.

Hopefully this saves many hours of frustrations for other people.
